### PR TITLE
Feat/modal

### PR DESCRIPTION
**Pull request by doylio**
## Changelog

- Create Modal component

## Demo

https://www.loom.com/share/9acd81a377da45158552eb3cbe04fdc5

### CODE REVIEW

File: src/molecules/Modal/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import React from "react";
+import ReactModal from "react-modal";
+import { useTheme } from "styled-components";
+
+import { ModalProps } from "./Modal.types";
+
+const Modal: React.FC<ModalProps> = ({
+    open,
+    color = "primary",
+    onOverlayClick,
```
Comment (dospore) [inline src/molecules/Modal/Modal.tsx:11]: thoughts on making this align with the props being passed into ReactModal and call it `onClose` or `onRequestClose`

Comment (doylio) [reply src/molecules/Modal/Modal.tsx]: Changed to `onRequestClose`

**Review by dospore — DISMISSED** [review]
lgtm

**Review by dospore — APPROVED** [review]
same same

**Pull request merged**